lang/cs/Org.Apache.REEF.IMRU.Examples/PipelinedBroadcastReduce/FaultTolerantPipelinedBroadcastAndReduce.cs

```diff
@@ -34,21 +35,45 @@ namespace Org.Apache.REEF.IMRU.Examples.PipelinedBroadcastReduce
     /// <summary>
     /// IMRU program that performs broadcast and reduce with fault tolerance.
     /// </summary>
-    public class FaultTolerantPipelinedBroadcastAndReduce : PipelinedBroadcastAndReduce
+    public sealed class FaultTolerantPipelinedBroadcastAndReduce : PipelinedBroadcastAndReduce
     {
         private static readonly Logger Logger = Logger.GetLogger(typeof(FaultTolerantPipelinedBroadcastAndReduce));
 
         [Inject]
-        protected FaultTolerantPipelinedBroadcastAndReduce(IIMRUClient imruClient) : base(imruClient)
+        private FaultTolerantPipelinedBroadcastAndReduce(IIMRUClient imruClient) : base(imruClient)
         {
         }
-        
+
+        /// <summary>
+        /// Runs the actual broadcast and reduce job with fault tolerance
+        /// </summary>
+        internal void Run(int numberofMappers, int chunkSize, int numIterations, int dim, int mapperMemory, int updateTaskMemory, int maxRetryNumberInRecovery, int totalAllowedFailurNumber)
+        {
+            var results = ImruClient.Submit<int[], int[], int[], Stream>(
+                new IMRUJobDefinitionBuilder()
+                    .SetMapFunctionConfiguration(BuildMapperFunctionConfig(maxRetryNumberInRecovery, totalAllowedFailurNumber))
+                    .SetUpdateFunctionConfiguration(UpdateFunctionConfig(numberofMappers, numIterations, dim))
```

Comment:
A lot of these `.Set` instructions are the same for regular and FT versions of `PipelinedBroadcastAndReduce`. Would it make sense to separate them into a method of `PipelinedBroadcastAndReduce` class, so that FT version calls that method + things unique for FT?